synapse/storage/databases/main/__init__.py

```diff
@@ -316,22 +316,25 @@ def get_users_paginate_txn(
             "get_users_paginate_txn", get_users_paginate_txn
         )
 
-    async def search_users(self, term: str) -> Optional[List[JsonDict]]:
+    async def search_users(self, term: str) -> List[Tuple[str, str, bool, bool, str]]:
```

Comment:
On SQLite I think these bools are actually going to be ints.

Reply:
Additionally:
- name is a nullable field (?!?!?!)
- so is password_hash
- so is user_type.

Reply:
With that said, on matrix.org:

```
matrix=> select count(*) from users;
  count   
══════════
 62887652
(1 row)

Time: 12499.198 ms (00:12.499)
matrix=> select count(*) from users where name IS NULL;
 count 
═══════
     0
(1 row)
```

Reply:
> On SQLite I think these bools are actually going to be ints.

Good catch.

> Additionally:

Yeah they're like all nullable for some reason. 😢

Reply:
Filed #16442 about `name` being nullable.

Reply:
I ignored `name` being nullable since we seem to not handle that anywhere.